business/authentication/auth_controller.go

```diff
@@ -17,6 +19,22 @@ type AuthController interface {
 	// other kind of error means that something unexpected happened.
 	Authenticate(r *http.Request, w http.ResponseWriter) (*UserSessionData, error)
 
+	// GetAuthCallbackHandler returns an http handler for authentication requests done to Kiali's web_root.
+	// This handler should determine if the request is an authentication request and perform
+	// authentication if needed. If it is not possible to reliably determine that a request is an
+	// authentication request, the fallbackHandler should be invoked to further handle the request.
+	//
+	// Usually, authentication requests done to the Kiali's web_root are because of callbacks of
+	// third-party authentication systems.
+	//
+	// This function can return a nil handler if the implementing AuthController does not handle
+	// authentication requests through the Kiali's web_root.
+	GetAuthCallbackHandler(fallbackHandler http.Handler) http.Handler
+
+	// PostRoutes adds any additional endpoints needed on the Kiali's router
+	// by the implementing AuthController.
+	PostRoutes(router *mux.Router)
+
```

Comment:
It might be better to restrict this interface to the set of methods that all implementations are expected to implement rather than add methods that will be noops for some implementations. These seem like they could be standalone functions exposed by the auth package outside of this interface.

Reply:
This might seem vague but it's harder for me to understand what an `AuthController` is/does with the added methods whereas before it was pretty clear. Said another way, adding routes to the router does not seem related to the rest of the `AuthController` interface. I'm not sure where the best place to put that logic is. Perhaps it could be an `authentication.PostRoutes(router *mux.Router)` method or something along those lines but you probably have a better feel for that than I do.

Reply:
> Another option is to do a type assertion on the interface

Yes, we can go that route... Sounds better to me.